### PR TITLE
Rewrite multitouch (breaking changes!)

### PR DESCRIPTION
**Pull request by LinusCDE**
Hi,
here is the rewrite for multitouch.

- The `MultitouchEvent::Touch` event was split into `Press` and `Move`
- Added `MultitouchEvent::Release`
- Events provide a finger struct
- State machine for finger recognition was rewritten
  - (multitouch recognition of independent 10 fingers was successfully tested)

---

**Note**:

Since the implementation now uses a `Mutex<FxHashMap>>` instead of atomic values, it is techicially a bit slower. Since the performance hit is really really tiny, I haven't fixed it (though `FxHashMap` is already faster than `HashMap` of std). Possible solutions would be:
 - Use an array to completely abandon hashing (the slots should be predictable anyway)
 - Use something like [evmap](https://lib.rs/crates/evmap) which would eliminate any performance hit caused by the mutex as well (though having many writes may not be the ideal usecase for this crate)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -43,12 +43,12 @@ extern crate ioctl_gen;
 
 extern crate aabb_quadtree;
 extern crate atomic;
+extern crate fxhash;
```
Comment (LinusCDE) [inline src/lib.rs:46]: This will probably also have an impact on PR #38 . @vmx

**Comment by fenollp** [comment]
10 fingers multitouch & tracking, that's nice!
IMO this should get in before #38 as its changes seem automatable enough.

**Comment by LinusCDE** [comment]
I also created a small [swipe-engine](https://github.com/LinusCDE/retris/blob/master/src/swipe.rs) that was aimed a my [tetris clone implementation](https://github.com/LinusCDE/retris/) for the reMarkable but may also be handy for other users (const variables probably need to be changed a bit though and made configurable).  
I'm fine if you just take that file and integrate it with any needed/wanted changes into this lib.

**Comment by vmx** [comment]
It's fine if this one merged before my #38, I can rebase. Though I'd prefer if I'd need to rebase only once. So it would be cool to merge things that should be merged first and then let me know when it's the turn for my PR so that I can then rebase it.

**Comment by canselcik** [comment]
This is awesome and as you all agreed, I'm merging this first so that #38 doesn't need to be rebased multiple times. Excellent work.

**Pull request merged**